languages/python/oso/polar/host.py

```diff
@@ -225,7 +225,8 @@ def replace_repr(match):
             instance_id = int(match[1])
             try:
                 instance = self.get_instance(instance_id)
-                return repr(instance)
+                unregistered = not self.ffi_polar.is_registered_constant(type(instance).__name__)
```

Comment:
I'm not sure you need to go to the core for this — host language is aware of registered types in the `Host.types` map (or in languages that haven't gotten the new DF stuff yet, might be stored in a `Host.classes` map or similar)

Reply:
of course! 🤦🏻 that is much much cleaner.